src/lib/negation-game/favor.ts

```diff
@@ -1,7 +1,32 @@
 export interface FavorArgs {
   cred: number;
   negationsCred: number;
+  restakeAmount?: number;
+  slashedAmount?: number;
+  doubtedAmount?: number;
+  totalRestakeAmount?: number;
 }
 
-export const favor = ({ cred, negationsCred }: FavorArgs) =>
-  cred > 0 ? Math.floor((100 * cred) / (negationsCred + cred)) : 0;
+export const favor = ({ 
+  cred, 
+  negationsCred,
+  restakeAmount = 0,
+  slashedAmount = 0,
+  doubtedAmount = 0
+}: FavorArgs) => {
+  let baseFavor;
+  if (cred === 0) {
+    baseFavor = 0;
+  } else if (negationsCred === 0) {
+    baseFavor = 100;
+  } else {
+    baseFavor = Math.floor(100 * cred / (cred + negationsCred));
+  }
```

Comment:
might sound nitpicky, but early returns always make the code way easier to understand:
https://williamdurand.fr/2013/06/03/object-calisthenics/#2-dont-use-the-else-keyword

Reply:
> To avoid early returns, pass, loops

did they ever justify that? 🤔 

functional maxxing or something?